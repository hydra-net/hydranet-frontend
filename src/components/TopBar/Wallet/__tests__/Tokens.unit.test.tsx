import { screen } from "@testing-library/react";

import { render } from "../../../../testUtils";
import { Tokens } from "../Token";

describe("<Tokens/>", () => {
  let container: HTMLElement;

  beforeEach(() => {
    const component = render(<Tokens />);
    container = component.container;
  });

  it("should render component", () => {
    expect(container).toMatchSnapshot();
  });

  it("should show default tokens", async () => {
    const HDXElement = await screen.findByText(/^HDX/i);
    const sHDXElement = await screen.findByText(/sHDX/i);
    const gHDXElement = await screen.findByText(/gHDX/i);

    expect(HDXElement).toBeInTheDocument();
    expect(sHDXElement).toBeInTheDocument();
    expect(gHDXElement).toBeInTheDocument();
  });
});
