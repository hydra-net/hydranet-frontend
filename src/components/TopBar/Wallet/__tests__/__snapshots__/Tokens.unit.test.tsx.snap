// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tokens/> should render component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-3"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-4"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <img
            height="28px"
            src="hdx.svg"
            style="margin-right: 8px;"
            width="28px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            HDX
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
            data-testid="balance-token"
            style="font-weight: 600;"
          >
            0.000000000
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-3"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-4"
      >
        <div
          class="MuiBox-root MuiBox-root-8"
        >
          <img
            height="28px"
            src="shdx.svg"
            style="margin-right: 8px;"
            width="28px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            sHDX
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
            data-testid="balance-token"
            style="font-weight: 600;"
          >
            0.000000000
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-3"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-4"
      >
        <div
          class="MuiBox-root MuiBox-root-10"
        >
          <img
            height="28px"
            src="ghdx.svg"
            style="margin-right: 8px;"
            width="28px"
          />
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            gHDX
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-11"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
            data-testid="balance-token"
            style="font-weight: 600;"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 50px;"
            />
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
